[HDInsight]Support new api version 2023-04-15-preview

yuc-Li opened the pull request:
### All SDK Contribution checklist:
Swagger PR link Azure/azure-rest-api-specs#24270

This checklist is used to make sure that common guidelines for a pull request are followed.

 - [ ] Please open PR in Draft mode if it is:
    - Work in progress or not intended to be merged.
    - Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] I have read the contribution guidelines.
- [ ] The pull request does not introduce breaking changes.
#### General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, see this page.
#### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
#### SDK Generation Guidelines
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The *.csproj and AssemblyInfo.cs files have been updated with the new version of the SDK. Please double check nuget.org current release version.
### Additional management plane SDK specific contribution checklist:
Note: Only applies to `Microsoft.Azure.Management.[RP] or Azure.ResourceManager.[RP]`

- [x] Include updated management metadata.
- [x] Update AzureRP.props to add/remove version info to maintain up to date API versions.
#### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add new service label and/or contact assigned reviewer.

- If the check fails at the Verify Code Generation step, please ensure:

  - Do not modify any code in generated folders.
  - Do not selectively include/remove generated files in the PR.
  - Do use generate.ps1/cmd to generate this PR instead of calling autorest directly.
Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
**Note: We have recently updated the PSH module called by generate.ps1 to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run**
following command:
  `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`
#### Old outstanding PR cleanup
Please note:
If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

Commit:
Update generate configration

Commit:
Regenerate source code.

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.HDInsight](https://apiview.dev/Assemblies/Review/5fef9d82db794df784880f01e1d8dc7c)

Commit:
Export-API

Commit:
regenerate source code [Sample].

Commit:
remove old test session records

Commit:
add change log

Commit:
Need a empty test-resources.json file here to ensure that the living test runs correctly.

Commit:
Add test case and update test session records

modify default location for cluster creation due to insufficient quotas in the eastus.

Commit:
modify default test location

Commit:
modift default test location

Commit:
Merge branch 'HDinsight_new_api_version_20230415' of https://github.com/yuc-Li/azure-sdk-for-net into HDinsight_new_api_version_20230415

Commit:
update changelog

Commit:
remove changelog unrelease section

Commit:
update test seesion record

Commit:
Manually modify the generate code

Commit:
Manual modification generate code

Commit:
Update generate code

Commit:
Merge branch 'HDinsight_new_api_version_20230415' of https://github.com/yuc-Li/azure-sdk-for-net into HDinsight_new_api_version_20230415

aim-for-better commented on sdk/hdinsight/Azure.ResourceManager.HDInsight/CHANGELOG.md:
```diff
@@ -1,14 +1,11 @@
 # Release History
 
-## 1.1.0-beta.2 (Unreleased)
+## 1.1.0-beta.2 (2023-06-16)
```
Can we remove the beta for this version? From my side, I hope after our test we can release one version without beta

yuc-Li replied:
Sure, updated.

Commit:
update changelog

Commit:
update csproj

ArthurMa1978 commented on eng/mgmt/mgmtmetadata/hdinsight_resource-manager.txt:
```diff
@@ -3,12 +3,12 @@ AutoRest installed successfully.
 Generating CSharp code
 Executing AutoRest command
-cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/hdinsight/resource-manager/readme.md --csharp --version=v2 --reflect-api-versions --csharp-sdks-folder=D:\source\azure-sdk-for-net\sdk
+cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/main/specification/hdinsight/resource-manager/readme.md --csharp --version=v2 --reflect-api-versions --csharp-sdks-folder=D:\source\azure-sdk-for-net\sdk
```
Track 2 dont use this file, please revert it

Commit:
revert git branch

ArthurMa1978 reviewed (changes requested):
Please fix the ci errors

ArthurMa1978 commented on sdk/hdinsight/Azure.ResourceManager.HDInsight/CHANGELOG.md:
```diff
@@ -1,14 +1,11 @@
 # Release History
 
-## 1.1.0-beta.2 (Unreleased)
+## 1.1.0 (2023-06-20)
```
Please provide a new date

yuc-Li replied:
Update. Please help release this[ PR](https://github.com/Azure/azure-sdk-for-net/pull/37199) to release version 1.1.0, and then release this version 1.1.1.

Commit:
Update release date

ArthurMa1978 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into HDinsight_new_api_version_20230415

ArthurMa1978 committed:
update

ArthurMa1978 committed:
Update

ArthurMa1978 committed:
Update

Commit:
Update csproj and release date

ArthurMa1978 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into HDinsight_new_api_version_20230415

ArthurMa1978 committed:
Merge branch 'HDinsight_new_api_version_20230415' of https://github.com/yuc-Li/azure-sdk-for-net into HDinsight_new_api_version_20230415

yuc-Li committed:
Merge branch 'main' into HDinsight_new_api_version_20230415

ArthurMa1978 committed:
Merge branch 'HDinsight_new_api_version_20230415' of https://github.com/yuc-Li/azure-sdk-for-net into HDinsight_new_api_version_20230415

ArthurMa1978 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into HDinsight_new_api_version_20230415
